dashboard/app/views/shared/_user_header.html.haml

```diff
@@ -33,6 +33,20 @@
       %span= t('nav.user.signin')
 
 :javascript
```

Comment:
Since the newly-added JS code modifies the DOM before the initial render (by swapping `'Hello User'` with `Hello [username]`), it's best to inline it as a script tag in the HTML directly after the element it manipulates. If we moved it into an existing JS-pipeline bundle we would get an unsightly flash of unfinished `'Hello User'` content on page load after the DOM is loaded and before the externally-referenced bundle finishes downloading/parsing/executing.

The existing code is mostly click handlers that don't affect the initial render, so it could be moved into an external JS pipeline without similar issues. The tradeoff though is that moving click-handler logic into a separate file decouples them from the elements their behavior affects, which could make the code more complicated.